modules/my_health/app/controllers/my_health/v1/messaging_preferences_controller.rb

```diff
@@ -17,6 +17,17 @@ def update
         render json: resource,
                serializer: MessagingPreferenceSerializer
       end
+
+      # Update preferredTeam value for a patient's list of triage teams
+      # @param updated_triage_teams - an array of objects with triage_team_id and preferred_team values
+      def update_triage_team_preferences
+        updated_triage_teams = Array(params[:updated_triage_teams])
+        sanitized_triage_teams = updated_triage_teams.map do |team|
+          team.permit(:triage_team_id, :preferred_team).to_h
+        end
+        resource = client.update_triage_team_preferences(sanitized_triage_teams)
+        render json: resource
```

Comment:
It might help to add the return type as comment to mention that you're only returning the status from the client call and not the full response.

Reply:
Also did you have any thoughts about using a begin/rescue to handle the errors? I could maybe see it being useful if you want to send back specific messaging for this error: 

```
'{"errorCode":170,"developerMessage":"","message":"At Least one Triage
        group must be set to Yes"}'
```

Reply:
I thought about handling that type of error. What stopped me was a thought that if that logic changes on MHV SM API end, then we will need to tweak the code again to stay in sync. While I do see a benefit of not sending a request knowing it will fail anyway 🤔

Reply:
Yeah that makes sense. It looks like this might be solved on the FE anyway actually? Unless I'm misunderstanding, the figma looks like it throws an error if one isn't selected. You could also do what you suggested and not make the request there too.
![Screenshot 2024-07-26 at 12 34 29 PM](https://github.com/user-attachments/assets/457caf23-9d46-4192-ba88-133d9d01a219)